chia/clvm/spend_sim.py

```diff
@@ -263,9 +264,10 @@ async def farm_block(
                     generator_bundle = bundle
                     return_additions = additions
                     return_removals = bundle.removals()
+                    spent_coins_ids = [r.name() for r in return_removals]
 
                     await self.coin_store._add_coin_records([self.new_coin_record(addition) for addition in additions])
-                    await self.coin_store._set_spent([r.name() for r in return_removals], uint32(self.block_height + 1))
+                    await self.coin_store._set_spent(spent_coins_ids, uint32(self.block_height + 1))
```

Comment:
I've struggled with the fact that the simulator doesn't use our normal code paths for running the chain. This was a particularly painful part, having to reach into an internal function of the coin store.